casper/src/main/rholang/WalletCheck.rho

```diff
@@ -0,0 +1,78 @@
+// WalletCheck: holds funds until a public key with the given hash is presented,
+//     then creates a wallet with that public key and those funds.
+
+// create: makes a wallet check
+//
+// bytearray hash: the hash of the public key
+// Purse purse: the funds to put in the wallet
+contract @["WalletCheck", "create"](@hash, @purse) = {
+  new doneStore in {
+    doneStore!(false) |
+    
+    // claims the wallet by presenting the preimage of the hash
+    //
+    // bytearray pubkey
+    // name statusOut
+    // bytearray sig: signature of [pubKey, statusOut]
+    contract @hash(@[pubKey, statusOut], @sig) = {
+      
+      // Make sure the purse hasn't already been claimed
+      for (@done <- doneStore) {
+        if (done) {
+          doneStore!(done) |
+          *statusOut!([false, "Already claimed wallet"])
+        } else {
+
+          // Verify signature 
+          new verifiedOut in {
+            @"secp256k1Verify"!([pubKey, statusOut].toByteArray(), sig, pubKey, *verifiedOut) |
+            for (@verified <- verifiedOut) {
+              if (verified) {
+                new hashOut in {
+
+                  // Check that given pubKey is the preimage of hash
+                  @"keccak256Hash"!(pubKey.toByteArray(), *hashOut) |
+                  for (@pkHash <- hashOut) {
+                    if (pkHash == hash) {
+                      new walletOut in {
+
+                        // Create the wallet
+                        @"BasicWallet"!(purse, "secp256k1", pubKey, *walletOut) |
+                        for (@wallet <- walletOut) {
+                          doneStore!(true) |
+                          
+                          // Install a forwarder to the wallet at a forgeable name.
```

Comment:
This is different than what I have done. I just put the wallet as a replicated send at a name related to the PK, so to use the wallet you would start with `for(@wallet <- rho:pubkey:...)` or something similar, and then you could call the wallet directly. I think is this a little cleaner than writing a forwarder for every method because it requires less boilerplate.

It is of course still valid in both approaches that we should use the registry when it be comes available.

Reply:
Done.